### ISSUE TITLE
[hold for #12775] The chat status is unread if the user pastes text on chat quickly before the chat opens up

### ISSUE THREAD

**Issue by kavimuru** [Expensify/App#17856]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:

1. Copy some text to your clipboard
2. Click on a chat and quickly press “ctrl + v” on your keyboard before the chat opens up
3. Observe the chat status on LHN and notice there is a new message indicator on the opened chat

## Expected result:
The chat status should not be unread and the notification badge should not show any unread messages.
## Actual result:
The chat status is unread and the notification badge shows the number of unread messages.
## Workaround:
Can the user still use Expensify without this being fixed? Have you informed them of the workaround?

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.4
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Notes/Photos/Videos:** Any additional supporting documentation

https://user-images.githubusercontent.com/43996225/233849943-3b4c171a-a1a3-41c5-8558-c1e8368661b7.mp4

https://user-images.githubusercontent.com/43996225/233849949-a609a197-cd2f-4fa1-ac20-f373eb68a3d4.mp4

**Expensify/Expensify Issue URL:**
**Issue reported by:** @nathan-mulugeta
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1681820237186739

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~018590d4659bae5feb</li>
        <li>Upwork Job ID: 1650599542417797120</li>
        <li>Last Price Increase: 2023-04-24</li>
    </ul>
</details>

**Comment by MelvinBot** [comment]
Triggered auto assignment to @joekaufmanexpensify (`Bug`), see https://stackoverflow.com/c/expensify/questions/14418 for more details.

**Comment by MelvinBot** [comment]
### Bug0 Triage Checklist ([Main S/O](https://stackoverflowteams.com/c/expensify/questions/14418))

- [ ] This "bug" occurs on a supported platform (ensure `Platforms` in OP are ✅)
- [ ] This bug is not a duplicate report (check [E/App issues](https://github.com/Expensify/App/issues) and [#expensify-bugs](https://expensify.slack.com/archives/C049HHMV9SM))
    - If it is, comment with a link to the original report, close the issue and add any novel details to the original issue instead
- [ ] This bug is reproducible using the reproduction steps in the OP. [S/O](https://stackoverflowteams.com/c/expensify/questions/14418#:~:text=The%20bug%20is%20reproducible)
    - If the reproduction steps are clear and you're unable to reproduce the bug, check with the reporter and QA first, then close the issue.
    - If the reproduction steps aren't clear and you determine the correct steps, please update the OP.
- [ ] This issue is filled out as thoroughly and clearly as possible
    - Pay special attention to the title, results, platforms where the bug occurs, and if the bug happens on staging/production.
- [ ] I have reviewed and subscribed to the linked Slack conversation to ensure Slack/Github stay in sync

**Comment by MelvinBot** [comment]
Job added to Upwork: https://www.upwork.com/jobs/~018590d4659bae5feb

**Comment by MelvinBot** [comment]
Current assignee @joekaufmanexpensify is eligible for the External assigner, not assigning anyone new.

**Comment by MelvinBot** [comment]
Triggered auto assignment to Contributor-plus team member for initial proposal review - @0xmiroslav (`External`)

**Comment by MelvinBot** [comment]
Triggered auto assignment to @puneetlath (`External`), see https://stackoverflow.com/c/expensify/questions/7972 for more details.

**Comment by hellohublot** [comment]
My guess is that the `order of socket updates is wrong`, the same as [here](https://github.com/Expensify/App/issues/16728#issuecomment-1520492441)


<img width="448" alt="Screen Shot 2023-04-25 at 11 12 59 AM" src="https://user-images.githubusercontent.com/20135674/234166048-b22f3870-f99a-4659-8b91-89d2ab418133.png">

**Comment by joekaufmanexpensify** [comment]
@hellohublot Interesting. So you're thinking this will also be fixed by https://github.com/Expensify/App/issues/12775 ?

**Comment by hellohublot** [comment]
@joekaufmanexpensify 

Actually, I think we should hold for #12775, #12775 can fix this

**Comment by joekaufmanexpensify** [comment]
Sounds good. @puneetlath @0xmiroslav y'all good with putting this on hold for that issue?

**Comment by puneetlath** [comment]
works for me!

**Comment by joekaufmanexpensify** [comment]
Once #12775 is complete, we'll test whether this is fixed. If so, we can close this out!

**Comment by joekaufmanexpensify** [comment]
Still on hold

**Comment by joekaufmanexpensify** [comment]
Same.

**Comment by joekaufmanexpensify** [comment]
Still on hold.

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Still on hold

**Comment by joekaufmanexpensify** [comment]
Same.

**Comment by puneetlath** [comment]
Going to un-assign myself for now, per the new process.

**Comment by joekaufmanexpensify** [comment]
Still on hold

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Still on hold

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Still on hold

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
same

**Comment by joekaufmanexpensify** [comment]
Still held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Still held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Same.

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
I just retested this and can no longer reproduce. Closing.
